.circleci/config.yml

```diff
@@ -1,28 +1,114 @@
 version: 2.1
-jobs:
-  lint:
+executors:
+  default:
+    working_directory: /go/src/github.com/mattermost/mattermost-push-proxy
     docker:
       - image: circleci/golang:1.12
 
-    working_directory: /go/src/github.com/mattermost/mattermost-push-proxy
+jobs:
+  lint:
+    executor:
+      name: default
     steps:
       - checkout
       - run: curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
       - run: make check-style
 
   test:
-    docker:
-      - image: circleci/golang:1.12
-
-    working_directory: /go/src/github.com/mattermost/mattermost-push-proxy
+    executor:
+      name: default
     steps:
       - checkout
       - run: curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
       - run: make test
 
+  build:
+    executor:
+      name: default
+    steps:
+      - checkout
+      - run: curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+      - run: make build-server
```

Comment:
I guess because you are running the test in another step you prefer this than executing `make dist` is that right?